tests/test_io_elementwise.py

```diff
@@ -53,6 +64,52 @@ def store(request, tmp_path) -> H5Group | ZarrGroup:
             file.close()
 
 
+sparse_formats = ["csr", "csc"]
+SIZE = 2500
+
+
+@pytest.fixture(params=sparse_formats)
+def sparse_format(request):
+    return request.param
+
+
+def create_dense_store(store, n_dims: int = 2):
+    X = np.random.randn(*((SIZE,) * n_dims))
+
+    write_elem(store, "X", X)
+    return store
+
+
+def create_sparse_store(sparse_format: Literal["csc", "csr"], store: G) -> G:
+    """Returns a store
+
+    Parameters
+    ----------
+    sparse_format
+    store
+
+    Returns
+    -------
+        A store with a key, `X` that is simply a sparse matrix, and `X_dask` where that same array is wrapped by dask
+    """
+    import dask.array as da
+
+    X = sparse.random(
+        SIZE,
+        SIZE,
```

Comment:
I think I remember where we had problems with shape being square. I think we had some tests which were bugged because they were accessing the shape of the wrong dimension to check the expected shape of the result. It doesn't matter if you do it access the wrong dimension when the arrays are square.

Can we just pass a shape argument in here, or use a non-square size?

Reply:
done with non-square shape